tests/common/fixtures/advanced_reboot.py

```diff
@@ -326,6 +335,35 @@ def __handleRebootImage(self):
         logger.info('Remove downloaded tempfile')
         self.duthost.shell('rm -f {}'.format(tempfile))
 
+    def __handleDockerImage(self):
+        """Download and install docker image to DUT
+        """
+        if self.new_docker_image is None:
+            return
+
+        for service_name in self.service_list:
+            data = {}
+            docker_image_name = self.duthost.shell('docker ps | grep {} | awk \'{{print $2}}\''.format(service_name))['stdout']
+            cmd = 'docker images {} --format \{{\{{.ID\}}\}}'.format(docker_image_name)
+            data['image_id'] = self.duthost.shell(cmd)['stdout']
+            data['image_name'], data['image_tag'] = docker_image_name.split(':')
+
+            local_image_path = '/tmp/{}.gz'.format(data['image_name'])
+            logger.info('Downloading new docker image for {} to {}'.format(service_name, local_image_path))
+            self.localhost.shell('curl {0}/{1}.gz --output {2}'.format(self.new_docker_image, data['image_name'], local_image_path), module_ignore_errors=True)
```

Comment:
Why downloading it here? sonic_installer upgrade-docker seems to accept URL

Reply:
Even if the URL of new_docker_image is provided by CLI option, it is not guarantee that each service has a new docker image available. So, here we allow user do this: user can provide new docker image for some of the services, those services will run in-service warm upgrade while other services will run in-service warm restart.